### DISCUSSION TITLE
A way for admins to do PLC operations without email

### DISCUSSION THREAD

**Discussion by mary-ext** [bluesky-social/atproto#2916] [Protocol (atproto)]
Right now with my self-hosted PDS, I'd have to configure SMTP just to receive emails containing verification codes, whether it's for email 2FA or doing PLC operations, what's more important is the latter. I'd prefer not configuring SMTP at all if I have to, if there was a fake SMTP server that doesn't involve setting up a bajillion things then maybe I would, but there doesn't seem to be any guidance on this.

Currently this seems to be my only blocker towards migrating my account, not an immediate blocker, but a blocker on "how do I move out of my PDS later on"

https://github.com/bluesky-social/atproto/blob/09656d6db548d18da88ff580aab70a848613584f/packages/pds/src/api/com/atproto/identity/requestPlcOperationSignature.ts#L33-L40

**Comment by codegod100** [comment]
xmpp server?

**Comment by itaru2622** [comment]
this matter also raises the difficulties on ops automation or end-to-end tests in some cases.
for example, when creating/enabling account for ozone/labeler, this plc operation required.
then it needs to made mail accessing code with passphrase or human opetations to get the plc token.
